pkg/cmd/server/origin/master.go

```diff
@@ -305,14 +296,30 @@ func (c *MasterConfig) InstallAPI(container *restful.Container) []string {
 	}
 
 	var root *restful.WebService
+	installed := 0
 	for _, svc := range container.RegisteredWebServices() {
 		switch svc.RootPath() {
 		case "/":
 			root = svc
 		case OpenShiftAPIPrefixV1Beta1:
 			svc.Doc("OpenShift REST API, version v1beta1").ApiVersion("v1beta1")
+
+			// add the current user filter
+			// TODO: factor this better
+			filter := currentUserContextFilter(c.getRequestsToUsers())
+			routes := svc.Routes()
+			for i := range routes {
+				route := &routes[i]
+				if route.Method == "GET" && (route.Path == OpenShiftAPIPrefixV1Beta1+"/ns/{namespace}/users/{name}" || route.Path == OpenShiftAPIPrefixV1Beta1+"/users/{name}") {
```

Comment:
Which highlights how abysmal our tests of auth and the master are right now.  Follow up item we need to come back to is that test-cmd needs to be validating auth, test-integration needs to be reusing the master code (not inventing its own), and this stuff needs to be unit tested more effectively.

----- Original Message -----
> Good catch, copy and paste fail.
> 
> ----- Original Message -----
> > >  	for _, svc := range container.RegisteredWebServices() {
> > >  		switch svc.RootPath() {
> > >  		case "/":
> > >  			root = svc
> > >  		case OpenShiftAPIPrefixV1Beta1:
> > >  			svc.Doc("OpenShift REST API, version v1beta1").ApiVersion("v1beta1")
> > > +
> > > +			// add the current user filter
> > > +			// TODO: factor this better
> > > +			filter := currentUserContextFilter(c.getRequestsToUsers())
> > > +			routes := svc.Routes()
> > > +			for i := range routes {
> > > +				route := &routes[i]
> > > +				if route.Method == "GET" && (route.Path ==
> > > OpenShiftAPIPrefixV1Beta1+"/ns/{namespace}/users/{name}" || route.Path ==
> > > OpenShiftAPIPrefixV1Beta1+"/users/{name}") {
> > 
> > I'm not seeing it.  Is currentUserContextFilter supposed to be checking for
> > a
> > '~' before doing a replacement?
> > 
> > ---
> > Reply to this email directly or view it on GitHub:
> > https://github.com/openshift/origin/pull/839/files#r24010928
>